examples/tensorflow/dist-mnist/Dockerfile

```diff
@@ -12,7 +12,13 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-FROM tensorflow/tensorflow:1.5.0
+FROM tensorflow/tensorflow:2.17.0
+
+# Using keras-2.17 because of bug on keras-3.4.1
+# which used by default by TF-2.17 (https://github.com/tensorflow/tensorflow/issues/72388)
```

Comment:
Could we reimplement the training code to be fully compatible with TF 2?

(Even If we can do it, I will not claim the reimplementing in this PR.)

Reply:
Tracking issue - #2319